backend/Controllers/community_controller.py

```diff
@@ -160,6 +161,15 @@ def list_communities(request):
             communitymember__user=user
         ).order_by("name")
 
+    # Check for filter phrase.
+    phrase = request.GET.get("phrase")
+    if phrase is not None:
+        if phrase != "" and not phrase.isspace():
+            for term in phrase.split():
+                communities = communities.filter(Q(name__icontains=term))
```

Comment:
Getting a NameError here, have you imported Q?

Reply:
Yes its imported on line 1 right?

Reply:
it works